File: contracts/enums/PersonalityVocabulary.ts
export interface PersonalityTermMapping {
  personalityName: string;
  semanticName: string;
  actionId: string;
  accessibilityLabel: string;
  category: string;
}

export const PersonalityVocabulary: readonly PersonalityTermMapping[] = [
  { personalityName: "Pop", semanticName: "Play", actionId: "media.pop", accessibilityLabel: "Play", category: "Playback" },
  { personalityName: "Puls", semanticName: "Pause", actionId: "media.puls", accessibilityLabel: "Pause", category: "Playback" },
  { personalityName: "Plop", semanticName: "Stop", actionId: "media.plop", accessibilityLabel: "Stop", category: "Playback" },
  { personalityName: "Pass", semanticName: "Next Track", actionId: "media.pass", accessibilityLabel: "Next Track", category: "Playback" },
  { personalityName: "Prior", semanticName: "Previous Track", actionId: "media.prior", accessibilityLabel: "Previous Track", category: "Playback" },
  { personalityName: "Pulse-Pass", semanticName: "Fast Forward", actionId: "media.pulse_pass", accessibilityLabel: "Fast Forward", category: "Playback" },
  { personalityName: "Pulse-Prior", semanticName: "Rewind", actionId: "media.pulse_prior", accessibilityLabel: "Rewind", category: "Playback" },
  { personalityName: "Pace", semanticName: "Playback Speed", actionId: "media.pace", accessibilityLabel: "Playback Speed", category: "Playback" },
  { personalityName: "Puff-Up", semanticName: "Fullscreen", actionId: "media.puff_up", accessibilityLabel: "Toggle Fullscreen", category: "Playback" },
  { personalityName: "Pacify", semanticName: "Mute", actionId: "media.pacify", accessibilityLabel: "Mute Audio", category: "Playback" },
  { personalityName: "Pumping-Slider", semanticName: "Volume Control", actionId: "media.volume", accessibilityLabel: "Volume Control", category: "Playback" },
  { personalityName: "Perpetual", semanticName: "Repeat", actionId: "media.perpetual", accessibilityLabel: "Repeat", category: "Playback" },
  { personalityName: "Potluck", semanticName: "Shuffle", actionId: "media.potluck", accessibilityLabel: "Shuffle", category: "Playback" },
  { personalityName: "Pump", semanticName: "Like", actionId: "social.pump", accessibilityLabel: "Like", category: "Social" },
  { personalityName: "Punch", semanticName: "Dislike", actionId: "social.punch", accessibilityLabel: "Dislike", category: "Social" },
  { personalityName: "Propel", semanticName: "Boost", actionId: "social.propel", accessibilityLabel: "Boost Creator", category: "Social" },
  { personalityName: "Pair", semanticName: "Subscribe", actionId: "social.pair", accessibilityLabel: "Subscribe", category: "Social" },
  { personalityName: "Part", semanticName: "Unsubscribe", actionId: "social.part", accessibilityLabel: "Unsubscribe", category: "Social" },
  { personalityName: "Packup", semanticName: "Playlist", actionId: "playlist.packup", accessibilityLabel: "Playlist", category: "Library" },
  { personalityName: "Pocket", semanticName: "Download", actionId: "storage.pocket", accessibilityLabel: "Download for Offline", category: "Storage" },
  { personalityName: "Pass-Pulse", semanticName: "Share", actionId: "social.pass_pulse", accessibilityLabel: "Share", category: "Social" },
  { personalityName: "Patter", semanticName: "Comments", actionId: "social.patter", accessibilityLabel: "Comments", category: "Social" },
  { personalityName: "Prattle", semanticName: "Live Chat", actionId: "social.prattle", accessibilityLabel: "Live Chat", category: "Social" },
  { personalityName: "Pinned-Patter", semanticName: "Pinned Comment", actionId: "social.pinned_patter", accessibilityLabel: "Pinned Comment", category: "Social" },
];

File: contracts/enums/PersonalityVocabulary.kt
package contracts.enums

import kotlinx.serialization.Serializable

/**
 * Mapping between Pulsy's energetic "P" brand personality vocabulary,
 * underlying semantic meaning, canonical Action IDs, and accessibility labels.
 *
 * Adheres strictly to ADR-008:
 * - Visual Personality communicates brand vibrancy.
 * - Action IDs remain deterministic and semantic.
 * - Accessibility labels remain standard and accessible to screen readers.
 */
@Serializable
data class PersonalityTermMapping(
    val personalityName: String,
    val semanticName: String,
    val actionId: String,
    val accessibilityLabel: String,
    val category: String
)

object PersonalityVocabulary {
    val TERMS = listOf(
        PersonalityTermMapping("Pop", "Play", ActionId.MEDIA_POP, "Play", "Playback"),
        PersonalityTermMapping("Puls", "Pause", ActionId.MEDIA_PULS, "Pause", "Playback"),
        PersonalityTermMapping("Plop", "Stop", ActionId.MEDIA_PLOP, "Stop", "Playback"),
        PersonalityTermMapping("Pass", "Next Track", ActionId.MEDIA_PASS, "Next Track", "Playback"),
        PersonalityTermMapping("Prior", "Previous Track", ActionId.MEDIA_PRIOR, "Previous Track", "Playback"),
        PersonalityTermMapping("Pulse-Pass", "Fast Forward", ActionId.MEDIA_PULSE_PASS, "Fast Forward", "Playback"),
        PersonalityTermMapping("Pulse-Prior", "Rewind", ActionId.MEDIA_PULSE_PRIOR, "Rewind", "Playback"),
        PersonalityTermMapping("Pace", "Playback Speed", ActionId.MEDIA_PACE, "Playback Speed", "Playback"),
        PersonalityTermMapping("Puff-Up", "Fullscreen", ActionId.MEDIA_PUFF_UP, "Toggle Fullscreen", "Playback"),
        PersonalityTermMapping("Pacify", "Mute", ActionId.MEDIA_PACIFY, "Mute Audio", "Playback"),
        PersonalityTermMapping("Pumping-Slider", "Volume Control", ActionId.MEDIA_VOLUME, "Volume Control", "Playback"),
        PersonalityTermMapping("Perpetual", "Repeat", ActionId.MEDIA_PERPETUAL, "Repeat", "Playback"),
        PersonalityTermMapping("Potluck", "Shuffle", ActionId.MEDIA_POTLUCK, "Shuffle", "Playback"),
        PersonalityTermMapping("Pump", "Like", ActionId.SOCIAL_PUMP, "Like", "Social"),
        PersonalityTermMapping("Punch", "Dislike", ActionId.SOCIAL_PUNCH, "Dislike", "Social"),
        PersonalityTermMapping("Propel", "Boost", ActionId.SOCIAL_PROPEL, "Boost Creator", "Social"),
        PersonalityTermMapping("Pair", "Subscribe", ActionId.SOCIAL_PAIR, "Subscribe", "Social"),
        PersonalityTermMapping("Part", "Unsubscribe", ActionId.SOCIAL_PART, "Unsubscribe", "Social"),
        PersonalityTermMapping("Packup", "Playlist", ActionId.PLAYLIST_PACKUP, "Playlist", "Library"),
        PersonalityTermMapping("Pocket", "Download", ActionId.STORAGE_POCKET, "Download for Offline", "Storage"),
        PersonalityTermMapping("Pass-Pulse", "Share", ActionId.SOCIAL_PASS_PULSE, "Share", "Social"),
        PersonalityTermMapping("Patter", "Comments", ActionId.SOCIAL_PATTER, "Comments", "Social"),
        PersonalityTermMapping("Prattle", "Live Chat", ActionId.SOCIAL_PRATTLE, "Live Chat", "Social"),
        PersonalityTermMapping("Pinned-Patter", "Pinned Comment", ActionId.SOCIAL_PINNED_PATTER, "Pinned Comment", "Social")
    )
}
